HathorNetwork/hathor-core
# build: allow disabling rate-limits in 'generate_nginx_config' command

This PR changes the nginx.conf generator script to allow disabling the rate-limits we include there.
It is supposed to be used only with internal use cases that are not exposed to the internet.

### What was done
- Added a new parameter `--disable-rate-limits` to the `generate_nginx_config` command in hathor cli
- When this parameter is enabled, we do not set rate-limits, and set higher connection limits in websocket endpoints.
- Included targets in Makefile to build the new docker image without rate-limits.

### How to test
Go into `extras/nginx_docker` and run `make docker-use-cases` to check the generated `nginx_use_cases.conf` file.

I've setup the old `make docker` target to now build both images, with and without rate-limits, using different tags.
Run it if you need to test the whole process, but be aware that it will push the images if you have enough permissions and are logged in.
